compiler/rustc_infer/src/infer/higher_ranked/mod.rs

```diff
@@ -24,9 +24,9 @@ impl<'a, 'tcx> CombineFields<'a, 'tcx> {
         // as-is, we need to do some extra work here in order to make sure
         // that function subtyping works correctly with respect to regions
         //
-        // Note: this is a subtle algorithm.  For a full explanation,
-        // please see the large comment at the end of the file in the (inlined) module
-        // `doc`.
+        // Note: this is a subtle algorithm.  For a full explanation, please see
+        // the rustc book:
```

Comment:
```suggestion
        // the rustc development guide:
```

The [rustc book](https://doc.rust-lang.org/rustc/) is a different book